src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.cs

```diff
@@ -93,8 +98,31 @@ internal abstract partial class AnalyzerAssemblyLoader : IAnalyzerAssemblyLoader
         /// </summary>
         private partial bool IsMatch(AssemblyName requestedName, AssemblyName candidateName);
 
+        private void CheckIfDisposed()
+        {
+#if NET
+            ObjectDisposedException.ThrowIf(_isDisposed, this);
+#else
+            if (_isDisposed)
+                throw new ObjectDisposedException(this.GetType().FullName);
+#endif
+        }
+
+        public void Dispose()
+        {
+            if (_isDisposed)
+                return;
+
+            _isDisposed = true;
+            DisposeWorker();
+        }
+
+        partial void DisposeWorker();
```

Comment:
```suggestion
        private partial void DisposeWorker();
```

This will force an implementation in `AnalyzerAssemblyLoader.Desktop.cs`. The content of that should be

```csharp
private void DisposeWorker() => EnsureResolvedUnhooked();
```

Reply:
done.